lib/Sema/CSSimplify.cpp

```diff
@@ -2943,6 +2943,11 @@ ConstraintSystem::matchFunctionIsolations(FunctionType *func1,
                                           ConstraintLocatorBuilder locator) {
   auto isolation1 = func1->getIsolation(), isolation2 = func2->getIsolation();
 
+  // An @isolated(any) function cannot be converted to a non-@isolated(any)
+  // synchronous function regardless of whether or not it is itself async.
+  if (isolation1.isErased() && !isolation2.isErased() && !func2->isAsync())
```

Comment:
the early exit here presumably means some of the 'score increase' logic that used to occur no longer will. does that seem generally okay?